pandas_gbq/gbq.py

```diff
@@ -1023,6 +1125,30 @@ def _generate_bq_schema(df, default_type='STRING'):
     return schema.generate_bq_schema(df, default_type=default_type)
 
 
+def wait_for_job(job, timeout_in_seconds=None):
+    # https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/bigquery/cloud-client/snippets.py
+    if timeout_in_seconds:
+        start = datetime.datetime.now()
+        timeout = start + datetime.timedelta(0, timeout_in_seconds)
+
+    with tqdm(
+        bar_format='Waiting for {desc} Elapsed: {elapsed}',
+        total=10000,
+    ) as progress:
+        while True:
```

Comment:
This loop can be modified to use `.result()` with a timeout.

Actually, I think we should refactor https://github.com/pydata/pandas-gbq/blob/master/pandas_gbq/gbq.py#L521-L539 to here, since the way we wait for a query job there is actually how we want to wait for any kind of job.